src/cow-react/modules/swap/pure/EthFlow/EthFlowStepper/steps/Step3.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import { Icon, Flag, Check } from 'react-feather'
+import styled from 'styled-components/macro'
+import { ExplorerLinkStyled, EthFlowStepperProps, SmartOrderStatus } from '..'
+import { StatusIconState } from '../StatusIcon'
+import { Step } from '../Step'
+
+const RefundMessage = styled.span`
+  color: #0d5ed9;
+`
+
+const ExpiredMessage = styled.span``
+
+export function Step3({ nativeTokenSymbol, tokenLabel, order, refund, cancelation }: EthFlowStepperProps) {
+  const { state, isExpired, rejectedReason } = order
+  const { isRefunded, refundTx } = refund
+  const { isCanceled, cancelationTx } = cancelation
+
+  const isIndexing = state === SmartOrderStatus.CREATION_MINED
+  const isIndexed = state === SmartOrderStatus.INDEXED
+  const isCreating = state === SmartOrderStatus.CREATING
+  const isFilled = state === SmartOrderStatus.FILLED
+
+  const isRefunding = !!refundTx && !isRefunded
+  const isCanceling = !!cancelationTx && !isCanceled
+
+  const expiredBeforeCreate = isExpired && (isCreating || isIndexing)
+
+  let label: string, stepStatus: StatusIconState, icon: Icon
+  if (expiredBeforeCreate) {
```

Comment:
function getStepConfig(tokenLabel: string, nativeTokenSymbol: string) {
    if (expiredBeforeCreate) {
        return {
            label: 'Receive ' + tokenLabel,
            status: 'pending',
            icon: 'Flag'
        }
    }

    if (isIndexing) {
        return {
            label: 'Receive ' + tokenLabel,
            status: 'not-started',
            icon: 'Flag'
        }
    }

    if (isFilled) {
        return {
            label: 'Received ' + tokenLabel,
            status: 'success',
            icon: 'Flag'
        }
    }

    ...
}